nexus/reconfigurator/planning/src/blueprint_editor/sled_editor/host_phase_2.rs

```diff
@@ -22,10 +22,19 @@ impl HostPhase2Editor {
         }
     }
 
-    pub fn set_value(&mut self, host_phase_2: BlueprintHostPhase2DesiredSlots) {
+    /// Set the host phase 2 information for this sled, returning the previous value.
+    pub fn set_value(
+        &mut self,
+        host_phase_2: BlueprintHostPhase2DesiredSlots,
+    ) -> BlueprintHostPhase2DesiredSlots {
         let BlueprintHostPhase2DesiredSlots { slot_a, slot_b } = host_phase_2;
+        let previous = BlueprintHostPhase2DesiredSlots {
+            slot_a: self.slot_a.value().clone(),
+            slot_b: self.slot_b.value().clone(),
+        };
         self.slot_a.set_value(slot_a);
```

Comment:
Should `ScalarEditor::set_value()` return the old value? (If it did, would that clean up this method a bit?)

Reply:
Ehh, I don't know if it's worth it. Up to you.